### ISSUE TITLE
Setting Suggestion Value from Object property

### ISSUE THREAD

**Issue by GiovanniL19** [aupac/ember-aupac-typeahead#14]
Hi,

I am unsure on how to set the suggestion value. For example 
![screen shot 2016-02-25 at 16 31 03](https://cloud.githubusercontent.com/assets/10454722/13326301/33ea38fe-dbdd-11e5-91a9-636d1d25b0ae.png)

The first result should say 'Test', the object contains a value called `companyName` which I would like to use.

My handlebar looks like this
`{{aupac-typeahead action=(action (mut typeAheadCase)) class='form-control' display=source.companyName source=assignToCases placeholder='Search for a case'}}`

**Comment by jackmatt2** [comment]
It need to reference a function, try this on your controller.

```
getDisplayKey(model) {
  return Ember.get(model, 'companyName');
}
```

And then set the display property to reference that function.

```
{{aupac-typeahead class="form-control" action=(action (mut typeAheadCase)) class='form-control' display=getDisplayKey source=assignToCases placeholder='Search for a case'}}
```

**Comment by GiovanniL19** [comment]
Thanks @jackmatt2 for your quick response! The function works great however the `display` is still showing the object :/

**Comment by jackmatt2** [comment]
Try supplying a custom suggestion template - inside the template you can do:

```
{{model.companyName}}
```

https://github.com/aupac/ember-aupac-typeahead#using-your-own-custom-template

**Comment by jackmatt2** [comment]
@GiovanniL19 is this issue solved?

**Comment by GiovanniL19** [comment]
@jackmatt2 sorry for late reply, been really busy! still no luck :/ I will probably start it again from scratch in a different project and see if the same issue happens

**Comment by GiovanniL19** [comment]
Solved! this is the code that worked for me:

```
assignToCases: function(query, syncResults, asyncResults) {
    var regex = new RegExp(`.*${query}.*`, 'i');
    var results = casesList.filter((item, index, enumerable) => {
      return regex.test(item.get('companyName'));
    })

    const nameList = [];
    results.forEach(function(result) {
      nameList.push(result.get('companyName'));
    });
    syncResults(nameList);
  }
```

Thanks @jackmatt2 for your help! :+1:

**Comment by initram** [comment]
I also have this problem, I would really like to avoid having to return a list of strings from the source function.

the issue can be seen here: https://ember-twiddle.com/9ce5aa66ba8908046044ac5dd7b68961?openFiles=controllers.application.js%2C

The suggestion text is displayed correctly in the search box, but not in the drop down.

Also: why is the scollbar always displayed?

**Comment by jackmatt2** [comment]
yes, this is quite annoying.  You can get around it in the suggestion by supplying a custom suggestion template but it is still displaying as `[object object]` in the text field once something is selected.

**Comment by initram** [comment]
Is there a reason why you always provide your own templates as default instead of relying on the default templates from typeahead? It seems that this is causing at least one of the issues seen here. You do not use the display function in your default suggestion template, but if you simply omitted the template then typeahead would supply a default template that does it correctly.

I am not sure about the issue with setting the text in the box when it is selected yet.

**Comment by initram** [comment]
I got it to work, but I don't think that it is pretty or clear that this is what is needed to support object suggestions:

https://ember-twiddle.com/9ce5aa66ba8908046044ac5dd7b68961?openFiles=controllers.application.js%2C

I had to set both the "display" and "transformSelection" (why does the component expose both? when would it make sense for them to be different?) to my function that pulls out the string, I also had to implement a function that can handle both strings and objects to be used as the action. As it will call it with the suggestion object first and the with the pure sting later.

**Comment by jackmatt2** [comment]
Agree, having two places for this makes no sense.
